js/base/Exchange.js

```diff
@@ -2069,6 +2069,32 @@ module.exports = class Exchange {
         return [ type, params ];
     }
 
+    handleSubTypeAndParams (methodName, market = undefined, params = {}) {
+        let subType = undefined;
+        // at first, check from market object
+        if (market !== undefined) {
+            if (market['linear']) {
+                subType = 'linear';
+            } else if (market['inverse']) {
+                subType = 'inverse';
+            }
+        }
+        // if it was not defined in market object
+        if (subType === undefined) {
+            const exchangeWideValue = this.safeString2 (this.options, 'defaultSubType', 'subType', 'linear');
+            const methodOptions = this.safeValue (this.options, methodName, {});
+            subType = this.safeString2 (methodOptions, 'defaultSubType', 'subType', exchangeWideValue);
+        }
+        // if set in params, it should override everything
+        const subTypeInParams = this.safeString2 (params, 'defaultSubType', 'subType');
+        // avoid omitting if it's not present
+        if (subTypeInParams !== undefined) {
```

Comment:
@ttodua, since "params" have priority over "options" shouldn't we start by checking them? Like we have to check them every time, but we don't need to check options if the information is available inside params, so I would suggest

```
if information in params:
    return information
else:
   check options
````
what do you think?

Reply:
ah this PR was meant to be closed, because Sam's PR was superior.

Reply:
no, wait, Sam's PR was different sorry. 
yes, you are right, better to save extra check/s